### PR TITLE
PLANET-6990 Apply the new style to body copy links

### PR DESCRIPTION
**Pull request by Osong-Michael**
### Description: [PLANET-6990](https://jira.greenpeace.org/browse/PLANET-6990)

**Testing**

Check this [post](https://www-dev.greenpeace.org/test-tavros/story/29458/peak-oil-decline-coronavirus-economy/) on the tavros instance. Because the _New Site Identity_ has been enabled, it reflects the new changes for body copy links.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [tavros](https://www-dev.greenpeace.org/test-tavros) | [admin](https://www-dev.greenpeace.org/test-tavros/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-tavros/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-tavros) | [composer-local.json](https://github.com/greenpeace/planet4-test-tavros/blob/main/composer-local.json)

:watch: 2023.03.22 11:31:23

File: assets/src/scss/new-identity/style.scss
Comment (GP-Dan-Tovbein) [inline assets/src/scss/new-identity/style.scss:16]: You can override the current CSS variable like [here](https://github.com/greenpeace/planet4-master-theme/pull/1959/files#diff-b418b2241107b99a37ed0291b0385d4ad8fe3762dd4d859be83bf83c070428d7R20). But not necessary if we will merge the other PR.

File: assets/src/scss/new-identity/style.scss
Comment (Osong-Michael) [reply assets/src/scss/new-identity/style.scss:42]: I added the `--link--color` variable, but it does not change anything, that is why i have specific styles for `pdf` and `external` links.

Comment (GP-Dan-Tovbein) [reply assets/src/scss/new-identity/style.scss:42]: I see it's being reflected
Screenshot 2023-03-27 at 11 55 50

Comment (GP-Dan-Tovbein) [reply assets/src/scss/new-identity/style.scss:42]: You'll also need to override the hover color for links. 
`--link--hover--color: var(--grey-900);`

File: assets/src/scss/new-identity/style.scss
Comment (GP-Dan-Tovbein) [inline assets/src/scss/new-identity/style.scss:37]: Related comment https://github.com/greenpeace/planet4-master-theme/pull/1955/commits/a777d07268c0665e4f824a71e301768912a1f6e3#r1149434376

File: assets/src/scss/new-identity/style.scss
Comment (GP-Dan-Tovbein) [inline assets/src/scss/new-identity/style.scss:49]: Checking the value of `offset-y`, currently set to `-22px`, I see that a bit inconsistent if you check that on screens since the different font sizes we have (see [here](https://github.com/greenpeace/planet4-master-theme/blob/main/assets/src/scss/base/_typography.scss#L11-L19)). You can also make that test and see that on small screens there is more gap over the text, but on large screens it's almost the same height.

Can we change the `offset-y` more dynamically? Also if the user override the font-size because of `font-size: var(--text--x-large-and-up--font-size, 1.25rem);`

If you think it's a bit complicated change we can follow up it afterwards and don't block the merge! 🙏 

What do you think?

Comment (Osong-Michael) [reply assets/src/scss/new-identity/style.scss:49]: Great idea, we can look into doing that dynamically. If you have any suggestions I will love to hear them.

File: assets/src/scss/new-identity/style.scss
Comment (GP-Dan-Tovbein) [reply assets/src/scss/new-identity/style.scss:44]: By doing this `--link--hover--text-decoration: none;`

File: assets/src/scss/new-identity/style.scss
Comment (mleray) [reply assets/src/scss/new-identity/style.scss]: We need to apply these new styles to all body links, not just the ones that have `external-link` or `pdf-link` classes

**Comment by mleray** [comment]
My previous comment still stands, we need to apply these changes to all body links and not just the external and PDF ones!

File: assets/src/scss/new-identity/style.scss
Comment (GP-Dan-Tovbein) [inline assets/src/scss/new-identity/style.scss:129]: Let's try something like this to style links without classes
```
a:where(.standalone-link, .external-link, .pdf-link),
a:not([class]) {
...
}
```

Comment (mleray) [reply assets/src/scss/new-identity/style.scss:129]: Before moving the styles to `theme.json` we used to define body link colors directly in [_typography.scss](https://github.com/greenpeace/planet4-master-theme/blob/main/assets/src/scss/base/_typography.scss#L102-L108) using just the `a` selector, so we should be able to do that now with the new identity but that might require us to add a few changes to other links that have other styles

File: assets/src/scss/new-identity/style.scss
Comment (GP-Dan-Tovbein) [inline assets/src/scss/new-identity/style.scss:130]: Looks good but honestly I see it very complex to maintain it afterwards.

But If this is the only way to lead with this scenario, then we can proceed to merge it and then think about a good solution for standardizing all the links.

Looking forward what @mleray thinks?

Comment (mleray) [reply assets/src/scss/new-identity/style.scss:130]: I think that we should just use the `a` selector without all the classes, and fix the few links that have different styles such as the contextual links, because having such a complicated selector is really not ideal... It's harder to understand, and difficult to maintain for us and to override for NROs!

File: assets/src/scss/new-identity/style.scss
Comment (Osong-Michael) [reply assets/src/scss/new-identity/style.scss:129]: Looking at the example page you created, one fix will be to add `.external-links` to the selector, another we can use the `post-details > a` selector, assuming all body copy links are present only within this context

Comment (mleray) [reply assets/src/scss/new-identity/style.scss:129]: Adding `.external-links` is definitely not the way to go, in my example the links are external but it could not be the case. If we do `.post-details > a` the ones in links won't be included, since they won't be direct children of the `.post-details` element. However we could maybe use `.post-details` 🤔 Something like `.post-details li > a`, however this will only work in Posts because other post types use different classnames (I think for Pages it's `page-content`?)

Comment (mleray) [reply assets/src/scss/new-identity/style.scss:129]: Maybe we should also use that for `p > a` actually, to make sure only the content is targeted and not other elements (footer, etc)

File: assets/src/scss/new-identity/style.scss
Comment (mleray) [inline assets/src/scss/new-identity/style.scss:130]: Actually, I see that for the [Lightbox](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/assets/src/components/Lightbox/setupLightboxForImages.js#L48) we use the `.post-content` selector rather than `.post-details` so maybe we could do that here too, for consistency?
Also, I think that maybe we can simplify this a bit by writing: 
```
.post-content,
.page-content {
  .standalone-link,
  p > a,
  li > a {
    ...
  }
}
```
What do you think? 🙂

Comment (mleray) [reply assets/src/scss/new-identity/style.scss:130]: Maybe it would also be nice to add a small comment like `// Body link` to clarify what this selector is for 🤔

**Review by mleray — APPROVED** [review]
Looks good to me 🎉 Just added one small comment but it's not a blocker, I'll leave it up to you 🙂

**Pull request merged**
